tests/test_diffusers.py

```diff
@@ -1170,3 +1172,566 @@ def test_stable_diffusion_xl_hpu_graphs(self):
 
         self.assertEqual(len(images), 10)
         self.assertEqual(images[-1].shape, (64, 64, 3))
+
+class GaudiStableDiffusionControlNetPipelineTester(TestCase):
+    """
+    Tests the StableDiffusionControlNetPipeline for Gaudi.
+    """
+
+    def get_dummy_components(self, time_cond_proj_dim=None):
+        torch.manual_seed(0)
+        unet = UNet2DConditionModel(
+            block_out_channels=(4, 8),
+            layers_per_block=2,
+            sample_size=32,
+            time_cond_proj_dim=time_cond_proj_dim,
+            in_channels=4,
+            out_channels=4,
+            down_block_types=("DownBlock2D", "CrossAttnDownBlock2D"),
+            up_block_types=("CrossAttnUpBlock2D", "UpBlock2D"),
+            cross_attention_dim=32,
+            norm_num_groups=1,
+        )
+
+        def init_weights(m):
+            if isinstance(m, torch.nn.Conv2d):
+                torch.nn.init.normal(m.weight)
+                m.bias.data.fill_(1.0)
```

Comment:
Is this specific to Gaudi?

Reply:
Not specific to gaudi. I have used control net init params from https://github.com/huggingface/diffusers/blob/main/tests/pipelines/controlnet/test_controlnet.py#L298